frontend/src/store/modules/auth/mutations.js

```diff
@@ -10,15 +12,18 @@ export default {
     },
     [USER_LOGOUT]: (state) => {
         state.token = undefined;
-        state.currentUser = {
-            name: '',
-            email: '',
-            id: null
-        };
+        state.currentUser = EMPTY_USER();
         state.isLoggedIn = false;
     },
     [SET_AUTHENTICATED_USER]: (state, user) => {
         state.isLoggedIn = true;
         state.currentUser = user;
+    },
+    [USER_UPDATE]: (state, response) => {
```

Comment:
you already have mutation `SET_AUTHENTICATED_USER`, also state doesn't have key `user`
In addition to that - you should never save password on frontend